### PR TITLE
codeintel: RFC 235: Create connection to codeintel database

### PR DESCRIPTION
**Pull request by efritz**
Steps toward [RFC 235: Move LSIF data to Postgres](https://docs.google.com/document/u/1/d/1Y9p29hK8xrPUTvBdWqP9uAHDg3JLV-HCaxsbCo9-YHQ). Closes https://github.com/sourcegraph/sourcegraph/issues/13884. 

This PR adds `CodeIntelPostgresDSN` to conftypes and connects to the configured DSN in the bundle manager and the worker. **Do not** merge this PR until https://github.com/sourcegraph/sourcegraph/issues/13883 has been completed and deployed.

### CODE REVIEW

File: enterprise/cmd/precise-code-intel-worker/main.go [hunk 1]
```diff
@@ -93,3 +96,19 @@ func mustInitializeStore() store.Store {
 
 	return store.NewWithHandle(basestore.NewHandleWithDB(dbconn.Global))
 }
+
+func mustInitializeCodeIntelDatabase() *sql.DB {
```
Comment (unknwon) [inline enterprise/cmd/precise-code-intel-worker/main.go:100]: nit: seems identical to the other `mustInitializeCodeIntelDatabase` in `enterprise/cmd/precise-code-intel-bundle-manager/main.go`?

**Review by unknwon — APPROVED** [review]

File: enterprise/cmd/precise-code-intel-worker/main.go (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/main.go:100]: I can take care of the duplication in a cleanup task. This is here just to ensure the configuration is correct on startup.

Comment (unknwon) [reply enterprise/cmd/precise-code-intel-worker/main.go:100]: SGTM!

**Review by gbrik — APPROVED** [review]
would be nice to clean up the way the prefix arguments work in `dbutil.PostgresDSN` and `dbconn.New` when the dust settles, but LGTM!

**Review by Strum355 — APPROVED** [review]

File: enterprise/cmd/precise-code-intel-worker/main.go [hunk 2]
```diff
@@ -93,3 +96,19 @@ func mustInitializeStore() store.Store {
...
+	postgresDSN := conf.Get().ServiceConnections.CodeIntelPostgresDSN
+	conf.Watch(func() {
+		if newDSN := conf.Get().ServiceConnections.CodeIntelPostgresDSN; postgresDSN != newDSN {
+			log.Fatalf("detected database DSN change, restarting to take effect: %s", newDSN)
+		}
+	})
+
+	db, err := dbconn.New(postgresDSN, "_codeintel")
+	if err != nil {
+		log.Fatalf("failed to connect to database: %s", err)
```
Comment (efritz) [inline enterprise/cmd/precise-code-intel-worker/main.go:110]: - [ ] add more specificity in these messages.

**Review by daxmc99 — APPROVED** [review]

**Comment by uwedeportivo** [comment]
LGTM

**Pull request merged**